### PR TITLE
fix(stdlib): validate url for sql.from/sql.to/socket.from

### PR DESCRIPTION
**Pull request by yzhang1991**
Now `sql.from()`, `sql.to()`, and `socket.from()` will use the URLValidator defined by the dependency interface to validate the URL that the user specified.

As part of the change, `secret.EmptySecretService{}` is now included in the default dependency setting. Some test harnesses are added to the `mock` and the `querytest` packages, respectively.
I am not sure if the `querytest` is the right package to put them, but I did not find a better place.

---
This change is [Reviewable](https://reviewable.io/reviews/influxdata/flux/1962)

### CODE REVIEW

File: mock/administration.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+import (
+	"context"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/memory"
+)
+
+// this struct is incomplete. You can add what you need when you need it.
+type Administration struct {
```
Comment (yzhang1991) [inline mock/administration.go]: This is to be used in the test for `sql.from` and `socket.from` tests.
For those data sources, the test is different from transformations like `kafka.to` and `sql.to`

File: stdlib/socket/from.go [hunk 2]
```diff
@@ -136,20 +136,25 @@ func createFromSocketSource(s plan.ProcedureSpec, dsid execute.DatasetID, a exec
...
-		url, err := neturl.Parse(spec.URL)
-		if err != nil {
-			return nil, err
-		}
-		scheme = url.Scheme
-		address = url.Host
-		if !contains(schemes, scheme) {
-			return nil, errors.Newf(codes.Invalid, "invalid scheme %s, must be one of %v", scheme, schemes)
-		}
+		// no scheme specified, use default scheme and use the entire url as address
```
Comment (yzhang1991) [inline stdlib/socket/from.go]: `URLValidator` needs to take a `*url.URL` parameter. So I just tweaked the logic here a little bit to let both branches converge to the validation phase.

File: mock/administration.go (hunk 1 above)
Comment (fchikwekwe) [reply mock/administration.go]: I think this may be a duplicated effort. I added a similar `administration.go` to the `mock` package.

Comment (fchikwekwe) [reply mock/administration.go]: Check the master branch to see if this is a similar addition.

File: stdlib/socket/from.go [hunk 3]
```diff
@@ -136,20 +136,25 @@ func createFromSocketSource(s plan.ProcedureSpec, dsid execute.DatasetID, a exec
...
+		spec.URL = schemes[0] + "://" + spec.URL
+	}
+	url, err := neturl.Parse(spec.URL)
```
Comment (fchikwekwe) [inline stdlib/socket/from.go]: Neutral is misspelled. Doesn't seem to be causing any errors, but you might wanna update the typo here.

File: querytest/source_validation.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	"github.com/influxdata/flux/plan"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+// Some sources are located by a URL. e.g. sql.from, socket.from
+// the URL/DSN supplied by the user need to be validated by a URLValidator{}
+// before we can establish the connection.
+// This struct (as well as the Run() method) acts as a test harness for that.
+type SourceUrlValidationTestCases []struct {
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply querytest/source_validation.go]: Can you create an issue to update `kafka.to` to use this test harness?

File: stdlib/socket/from_private_test.go [hunk 5]
```diff
@@ -0,0 +1,51 @@
+package socket
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply stdlib/socket/from_private_test.go]: @affo I respectfully disagree. The need to test a function is not a good reason to make the methods public.  If the other methods in other source/transformations are only public so we can test them then we should make them private and update the tests. That can come later, it doesn't need to be part of this PR.

File: stdlib/sql/from_private_test.go [hunk 6]
```diff
@@ -0,0 +1,74 @@
+package sql
+
+import (
+	"github.com/influxdata/flux/dependencies/url"
+	"github.com/influxdata/flux/querytest"
+	"testing"
```
[the remark this answers is not in the document]
Comment (nathanielc) [reply stdlib/sql/from_private_test.go]: On these kinds of issues you should be able to configure your IDE to do this for you.

File: stdlib/socket/from_private_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (affo) [reply stdlib/socket/from_private_test.go]: I see your point and that is absolutely reasonable 👍 

I guess that they are public for that precise reason.

File: stdlib/sql/source_validator.go [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+		u, err = neturl.Parse(dataSourceName)
+		if err != nil {
+			return errors.Newf(codes.Invalid, "invalid data source url: %v", err)
+		}
+	default:
+		return errors.Newf(codes.Invalid, "sql driver %s not supported", driverName)
+	}
+
+	if err = validator.Validate(u); err != nil {
+		return errors.Newf(codes.Invalid, "data source did not url pass validation: %v", err)
```
[the remark this answers is not in the document]
Comment (yzhang1991) [reply stdlib/sql/source_validator.go]: Good catch!

File: mock/administration.go (hunk 1 above)
Comment (yzhang1991) [reply mock/administration.go]: I have merged the master into my branch and resolved the conflicts. the table_test refactor and the elimination of the MockAllocator will come up soon.

File: querytest/source_validation.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (yzhang1991) [reply querytest/source_validation.go]: I will do that quickly in a separate PR.

File: stdlib/socket/from_private_test.go [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+import (
+	"github.com/influxdata/flux/dependencies/url"
+	"github.com/influxdata/flux/querytest"
+	"testing"
```
[the remark this answers is not in the document]
Comment (yzhang1991) [reply stdlib/socket/from_private_test.go]: Fixed.

File: stdlib/sql/from.go [hunk 9]
```diff
@@ -3,8 +3,6 @@ package sql
 import (
 	"context"
 	"database/sql"
-
-	_ "github.com/go-sql-driver/mysql"
 	"github.com/influxdata/flux"
```
[the remark this answers is not in the document]
Comment (yzhang1991) [reply stdlib/sql/from.go]: Fixed.

File: stdlib/sql/from_private_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (yzhang1991) [reply stdlib/sql/from_private_test.go]: Fixed.

File: stdlib/sql/to_test.go [hunk 10]
```diff
@@ -1,13 +1,16 @@
 package sql_test
 
 import (
+	"github.com/influxdata/flux/dependencies/dependenciestest"
+	"github.com/influxdata/flux/dependencies/url"
 	"github.com/influxdata/flux/plan"
+	"github.com/stretchr/testify/assert"
```
[the remark this answers is not in the document]
Comment (yzhang1991) [reply stdlib/sql/to_test.go]: Fixed.

File: querytest/source_validation.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (yzhang1991) [reply querytest/source_validation.go]: https://github.com/influxdata/flux/pull/1989

File: stdlib/experimental/bigtable/bigtable_test.go [hunk 11]
```diff
@@ -8,6 +8,8 @@ import (
 
 	"cloud.google.com/go/bigtable"
 	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/assert"
+
```
Comment (affo) [inline stdlib/experimental/bigtable/bigtable_test.go]: fix imports

File: stdlib/sql/from.go [hunk 12]
```diff
@@ -4,15 +4,15 @@ import (
 	"context"
 	"database/sql"
 
-	_ "github.com/go-sql-driver/mysql"
+	_ "github.com/lib/pq"
+
```
Comment (affo) [inline stdlib/sql/from.go]: fix imports.

You should only separate stdlib VS others.
Don't know if this is widespread, I know this is our practice :)

File: stdlib/sql/to_test.go [hunk 13]
```diff
@@ -1,13 +1,19 @@
...
+
+	"github.com/influxdata/flux/dependencies/dependenciestest"
+	"github.com/influxdata/flux/dependencies/url"
+	"github.com/influxdata/flux/plan"
+
 	"github.com/DATA-DOG/go-sqlmock"
 	"github.com/google/go-cmp/cmp"
 	"github.com/google/go-cmp/cmp/cmpopts"
-	flux "github.com/influxdata/flux"
+
```
Comment (affo) [inline stdlib/sql/to_test.go]: fix imports

Comment (yzhang1991) [reply stdlib/sql/to_test.go]: Why is this import necessary?

**Pull request merged**